benchmarks/README.md

```diff
@@ -0,0 +1,74 @@
+# README
+
+## Overview
+We use `asv` to benchmark some representative brainglobe workflows. The `asv` workflow is roughly as follows:
+1. `asv` creates a virtual environment (as defined in the `asv.conf.json` file).
+1. It installs the version of the `brainglobe-workflows` software package corresponding to the tip of the locally checked-out branch.
+1. It runs the benchmarks defined (locally) under `brainglobe-workflows/benchmarks/benchmarks` and saves the results to `brainglobe-workflows/benchmarks/results` as json files.
+1. With `asv publish`, the output json files are 'published' into an html directory (`brainglobe-workflows/benchmarks/html`)
+1. With `asv preview` the html directory can be visualised in a static site.
+
+
+There are three main ways in which these benchmarks can be useful to developers:
+1. Developers can run the available benchmarks locally [on a small test dataset](#running-benchmarks-locally-on-default-small-dataset).
+1. Developers can run these benchmarks on [data they have stored locally](#running-benchmarks-locally-on-custom-data).
+1. We also plan to run the benchmarks internally on a large dataset, and make the results publicly available.
+
+## Installation
+To [install asv](https://asv.readthedocs.io/en/stable/installing.html):
+```
+pip install asv
+```
+Note that to run the benchmarks you only need to have `asv` in your virtual environment. You _do not_ need to install a development version of `brainglobe-workflows`, since `asv` will create a separate Python virtual environment to run the benchmarks on it. However, for convenience we include `asv` as part of the `[asv]` dependencies, so an environment with `brainglobe-workflows[dev]` can be used to run the benchmarks.
+
+
+## Running benchmarks on a default small dataset
+
+
+1. Git clone the `brainglobe-workflows` repository
+    ```
+    git clone https://github.com/brainglobe/brainglobe-workflows.git
+    ```
+2. Install the `asv` version of the `brainglobe-workflows` package:
+    ```
+    pip install .[asv]
+    ```
+    This is mostly for convenience: the `[asv]` specification includes `asv` as a dependency, but to run the benchmarks it would be sufficient to use an environment with `asv` only. `asv` creates its own virtual environment for the benchmarks, where it builds and installs the relevant version of the `brainglobe-workflows` package. By default, the version at the tip of the currently checked out branch is installed.
+3. Launch the benchmarks, by running from the directory where `asv.conf.json` is at:
+    ```
+    asv run
+    ```
+    This will run the local benchmarks, with the default config at `brainglobe_workflows/configs/cellfinder.json`, on a small dataset downloaded from [GIN](https://gin.g-node.org/G-Node/info/wiki).
```

Comment:
Shouldn't this readme be more about general benchmarks, with cellfinder being one package that can be benchmarked? Currently, this readme will need to be rewritten when we have more benchmarks.